pyproject.toml

```diff
@@ -31,13 +31,40 @@ exclude = [
   "requirements-dev.txt",
 ]
 
+[tool.uv]
+dev-dependencies = [
+    "covdefaults>=2.3.0",
+    "duckdb>=1.1.1",
+    "hypothesis>=6.112.1",
+    "pandas>=2.0.3",
+    "polars>=1.8.1",
+    "pre-commit>=3.5.0",
+    "pyarrow>=17.0.0",
+    "pytest>=8.3.3",
+    "pytest-cov>=5.0.0",
+    "pytest-env>=1.1.5",
+    "pytest-randomly>=3.15.0",
+    "scikit-learn>=1.3.2",
+    "tqdm>=4.66.5",
+    "typing-extensions>=4.12.2",
+    "dask[dataframe]>=2023.5.0",
+]
+
 [project.optional-dependencies]
-cudf = ["cudf>=23.08.00"]
+docs = [
+    "jinja2>=3.1.4",
+    "markdown-exec[ansi]>=1.9.3",
+    "mkdocs>=1.6.1",
+    "mkdocs-material>=9.5.36",
+    "mkdocstrings[python]>=0.26.1",
+    "polars>=1.0.0",
+]
+#cudf = ["cudf>=23.08.00"]
 modin = ["modin"]
 pandas = ["pandas>=0.25.3"]
 polars = ["polars>=0.20.3"]
 pyarrow = ["pyarrow>=11.0.0"]
-dask = ["dask[dataframe]>=2024.7"]
+dask = ["dask[dataframe]>=2023.5.0"]
```

Comment:
Maybe Marco recalls why that was the min version

Reply:
Not sure why this was resolved, but 2024.7 is indeed the min version we can support, as we need dask_expr.